storm-core/src/jvm/org/apache/storm/grouping/LoadAwareShuffleGrouping.java

```diff
@@ -36,7 +37,7 @@
 
     @Override
     public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<Integer> targetTasks) {
-        random = new Random();
+        random = new XORShiftRandom();
```

Comment:
Thread safety here is nice, but not a requirement.  If we double up on a few routs it is probably not that big of a deal.